torch/csrc/api/src/nn/modules/linear.cpp

```diff
@@ -49,5 +53,37 @@ Tensor LinearImpl::forward(const Tensor& input) {
   AT_ASSERT(!options.with_bias() || bias.defined());
   return torch::linear(input, weight, bias);
 }
+
+BilinearOptions::BilinearOptions(int64_t in1_features, int64_t in2_features, int64_t out_features) : in1_features_(in1_features), in2_features_(in2_features), out_features_(out_features) {}
+
+BilinearImpl::BilinearImpl(BilinearOptions options) : options(options) {
+  reset();
+}
+
+void BilinearImpl::reset() {
+  weight =
+      register_parameter("weight", torch::empty({options.out_features_, options.in1_features_, options.in2_features_}));
```

Comment:
`options.name_` are private now and we would need to change it to `options.name()`. (For example, `options.out_features_` would be `options.out_features()`.)